### PR TITLE
separate_functions

### PR DESCRIPTION
**Pull request by crewtool**
# Description
This PR logically splits some functions related to grid into few files in newly created `grid` folder:
- infrastructure contains defualt values on the clusters
- prepare_configs contains loading and merging configs from multiple sources
- setup_arguments contains setup arugments related function

It depends on https://github.com/llm-random/llm-random/pull/368

# Checklist
   - [ ] Is this PR focused on a single feature without the possibility of being divided into smaller PRs?
   - [ ] Are all variables set with logical defaults that are backward compatible?
   - [ ] I attached link to neptune if it was necessary
   - [ ] Have you successfully executed the linter and tests?
   - [ ] Are all functions concise and don't require splitting?
   - [ ] Is there documentation for arguments and complex logic?
   - [ ] Are there no temporary solutions in the code? Should a task be added to Trello to refine the code?
   - [ ] Are all functions placed in appropriate files?
   - [ ] Is the PR name meaningful and descriptive?
   - [ ] Is PR description provided, or unnecessary?
  
Other questions:
   - [ ] If I made changes to the requirements or anything related to the Singularity image I did generate new image and upload a new image to the clusters
   - [ ] If this change is significant enough I notified all individuals working with this repository (@channel in slack channel)
   - [ ] If there is a need for a second reviewer I requested additional review 

# Reviewer's checklist:
   - [ ] Does the Neptune link work and does it actually compare the situation before and after the PR?
   - [ ] Do I understand the code?
   - [ ] Do I think another reviewer is not needed or I commented that this PR need another reviewer?

### CODE REVIEW

File: lizrd/grid/prepare_configs.py [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+        split.append((infrastructure_config, training_config))
+    return split
+
+
+def check_interactive_debug_not_in_further_configs(configs: List[dict]):
+    options = [config.get("interactive_debug", False) for config in configs[1:]]
+    assert not any(options), "interactive_debug can only be set in the first config"
+
+
+def prepare_configs(
```
Comment (crewtool) [inline lizrd/grid/prepare_configs.py:74]: Crucial function here ^^^

File: lizrd/scripts/grid_utils.py [hunk 2]
```diff
@@ -253,42 +189,11 @@ def get_train_main_function(runner: str):
...
-        "hf_datasets_cache": HF_DATASETS_CACHE,
-        "singularity_image": SINGULARITY_IMAGE,
-        "runs_multiplier": RUNS_MULTIPLIER,
-    }
-    return setup_args
-
-
 def translate_to_argparse(param_set: dict):
     runner_params = []
+    sorted_param_set = dict(sorted(param_set.items()))
```
Comment (Krystian-Krol) [inline lizrd/scripts/grid_utils.py:194]: Use `OrderedDict` here as dict don't keep the input order

File: lizrd/scripts/grid_utils.py [hunk 3]
```diff
@@ -306,116 +211,37 @@ def translate_to_argparse(param_set: dict):
...
-            parser = blanks_introduce_parser_arguments(parser)
+    for setup_args, trainings_args in grid:
+        for i, training_args in enumerate(trainings_args):
+            runner_params = translate_to_argparse(training_args)
+            runner = setup_args["runner"]
+            parser = argparse.ArgumentParser()
+            if runner == "research.conditional.train.cc_train":
+                parser = cc_introduce_parser_arguments(parser)
+            else:
+                parser = blanks_introduce_parser_arguments(parser)
```
Comment (Krystian-Krol) [inline lizrd/scripts/grid_utils.py:223]: @maciek-pioro `blanki` się rozlewają

**Review by Krystian-Krol — COMMENTED** [review]
Mam takie dwie uwagi, reszta wygląda spoko jak na kod w którym i tak zaraz zmieni się struktura. 
Trochę duży ten commit więc przydałaby się druga para oczu do acceptu

File: lizrd/grid/prepare_configs.py [hunk 4]
```diff
@@ -0,0 +1,109 @@
...
+    return hash_md5
+
+
+def recursive_update(base_dict, update_dict):
+    for key, value in base_dict.items():
+        if isinstance(value, dict):
+            update_dict[key] = recursive_update(value, update_dict.get(key, {}))
+        elif key not in update_dict:
+            update_dict[key] = value
+    return update_dict
```
Comment (Krystian-Krol) [inline lizrd/grid/prepare_configs.py:20]: I do believe it should be the other way around `base` is what should be modified using `update`, so update updates base. At least it's how I understand this naming. Please rename, for example `patch_dict` and `base_dict` or whatever you think makes sense here

File: lizrd/grid/prepare_configs.py [hunk 5]
```diff
@@ -0,0 +1,109 @@
...
+) -> List[Tuple[dict, dict]]:
+    configs, all_config_paths = load_with_inheritance(config_path)
+
+    for config in configs:
+        config["params"]["git_branch"] = git_branch
+        config["params"]["path_to_entry_config"] = config_path
+        config["params"]["all_config_paths"] = ",".join(all_config_paths)
+
+    for config in configs:
+        defualt_params = prepare_default_infrastructure_params(
```
Comment (Krystian-Krol) [inline lizrd/grid/prepare_configs.py:85]: make ctrl+shift+F over `defualt` as it's yet another typo

**Review by Krystian-Krol — CHANGES_REQUESTED** [review]
Made one more pass

File: lizrd/grid/prepare_configs.py (hunk 4 above)
Comment (crewtool) [reply lizrd/grid/prepare_configs.py:20]: To pisane przez Szymona, na inną PR

Comment (Krystian-Krol) [reply lizrd/grid/prepare_configs.py:20]: i tak robisz małą zmianę stylistyczną (zmiana miejsca tej funkcji), więc zmiana nazwy lokalnych parametrów (tylko w ramach tych 6 linijek) nie jest zmianą wymagającą osobnej PR moim zdaniem

File: lizrd/grid/prepare_configs.py (hunk 5 above)
Comment (crewtool) [reply lizrd/grid/prepare_configs.py]: done

File: lizrd/grid/prepare_configs.py (hunk 4 above)
Comment (crewtool) [reply lizrd/grid/prepare_configs.py:20]: https://github.com/llm-random/llm-random/pull/374 dodałem osobną PR

**Review by maciek-pioro — APPROVED** [review]

**Review by Krystian-Krol — APPROVED** [review]

**Pull request merged**